src/cli/cli.cpp

```diff
@@ -2681,13 +2689,49 @@ template <> otError Interpreter::Process<Cmd("csl")>(Arg aArgs[])
         error = ProcessSet(aArgs + 1, otLinkSetCslTimeout);
     }
     else
-    {
-        error = OT_ERROR_INVALID_ARGS;
-    }
+#endif // OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+        /**
```

Comment:
These seem to be included unconditionally (no `#if` checks).

How about we move this whole block to the top of the method? We could start with these unconditional items, then fall into the `else if` blocks (which have the `#if` checks), and then end everything with this final `else`:

```c++
        else
        {
            error = OT_ERROR_INVALID_ARGS;
        }
```

Would this help avoid a potential hanging `else` issue with the earlier statements?